### ISSUE TITLE
Missing nested model in Model Schema

### ISSUE THREAD

**Issue by lnowak** [swagger-api/swagger-ui#1203]
My service description looks like this:

    {  
      "apiVersion":"1.0.0",
      "swaggerVersion":"1.2",
      "basePath":"http://localhost:8000/api",
      "resourcePath":"/authentication",
      "produces":[  
        "application/json"
      ],

      "apis":[  
      {  
         "path":"/authentication/checkToken",
         "operations":[  
            {  
               "method":"GET",
               "summary":"Checks if the authentication token is valid",
               "notes":"",
               "type":"response",
               "nickname":"checkToken",
               "parameters":[  
                  {  
                     "name":"token",
                     "required":false,
                     "type":"string",
                     "paramType":"query",
                     "allowMultiple":false
                  }
               ]
            }
         ]
      }
    ],
    "models":{  
      "status":{  
         "id":"status",
         "description":"Status of the operation",
         "required":[  
            "errorCode"
         ],
         "properties":{  
            "errorCode":{  
               "type":"integer",
               "format":"int32",
               "description":"error code identifier"
            },
            "message":{  
               "type":"string",
               "description":"error code message"
            }
         }
      },
      "response":{  
         "id":"response",
         "description":"Wrapper object for status of the operation",
         "required":[  
            "status"
         ],
         "properties":{  
            "status":{  
               "$ref":"status",
               "description":"status of the operation"
             }
           }
        }
      }
    }

I have noticed two problems:
1. Model is incorrect - status object is optional and status definition is missing,
2. Model Schema is incorrect - nested Model status is not attached.
![model](https://cloud.githubusercontent.com/assets/1785411/7299259/1713a68e-e9d5-11e4-9f22-e3bb1885e16c.jpg)
![model schema](https://cloud.githubusercontent.com/assets/1785411/7299262/1d0582ba-e9d5-11e4-8ba9-3ff54ca5e695.jpg)

**Comment by ponelat** [comment]
problem one is a duplicate of #1152,
looking into problem two...

**Comment by ponelat** [comment]
Ok problem two, where nested models were not getting shown has been solved in the latest source.
Problem one, where all paramters are shown as *optional* is still apparent, @whitlockjc any news on #1152?

Note: latest source is compiled from
swagger-api/swagger-ui@d76a62caf7e25115921db44accc61843de1727b2
swagger-api/swagger-js@85caf4c95772ada486181ff4ba7a1d037860723d

**Comment by stevage** [comment]
I think I have the same issue:
```
definitions:
  timetable_value:
    type: object
    properties:
      time_table_utc:
        type: string
        format: date
      time_realtime_utc:
        type: string
        format: date
      flags:
        type: string

paths:
...
      responses:
        200:
          schema:
            title: Departures
            type: array
            description: "..."
            items:
              - $ref: "#/definitions/timetable_value"
```

**Comment by ponelat** [comment]
@stevage I think the fix is in the latest master. We just released a new version. Try with that, and let us know :) Also if it is a problem, let us know what's actually wrong... screenshot + spec is best.

**Comment by stevage** [comment]
Ok, well I don't know if this is the exact same issue or not, but it's still present even with latest master.

In [this version](https://github.com/OKFNau/PTV-API-Swagger/blob/403011e3405d612ad2da2983b4e2b43c1225a8ce/swagger.json) the response is an array of items referred to by `$ref`. 

No model schema:

![screenshot 2015-05-24 21 55 43](https://cloud.githubusercontent.com/assets/678344/7787750/f8972962-025f-11e5-8de6-7f31e2310591.png)

If I [duplicate the definition and remove the `$ref`](https://github.com/OKFNau/PTV-API-Swagger/commit/334a0149cc8b77a28e50975c9e6be6d0ef9f4b0a), the model schema displays correctly:

![screenshot 2015-05-24 21 55 30](https://cloud.githubusercontent.com/assets/678344/7787757/182fac0e-0260-11e5-87a2-a51d4e0971ed.png)

But the extra downside is the "Model" tab now says "Inline model 1".

![screenshot 2015-05-24 21 59 47](https://cloud.githubusercontent.com/assets/678344/7787760/36532a26-0260-11e5-8761-a2dcc0c8b2e4.png)

**Comment by stevage** [comment]
Hmm, so partially I had an error in my code. My JSON for the `items` property contained an array (I always get tripped up by the leading `-` in YAML)

But it looks like `Model Schema` only generates shows code in this situation when explicitly fed `example` values. Is this the expected behaviour? For instance:

          schema:
            title: Departures
            type: array
            items:
              $ref: "#/definitions/timetable_value"
            example:
              time_table_utc: "2015-04-23Z18:00"
              time_realtime_utc: "2015"
              flags: ""
              stop:
                suburb: "Belgrave"
                transport_type: "train"

Shows:
```
{
  "time_table_utc": "2015-04-23Z18:00",
  "time_realtime_utc": "2015",
  "flags": "",
  "stop": {
    "suburb": "Belgrave",
    "transport_type": "train"
  }
}
```

That seems a bit odd, but at least I can get output. Am I misunderstanding?

**Comment by ponelat** [comment]
@stevage could you rephrase a little? This has high priority, so I'd like to make sure I understand correctly. From what I can tell, you don't get any content under "Model Schema" if you don't explicitly set `example`. What *do* you get... without `example`? Thanks

**Comment by stevage** [comment]
Without example, there is nothing present at all under Model Schema, as in the first screenshot.

Sorry for the unclear wording - it's hard to be clear when you're still learning the tool and don't quite understand what you're doing. But to restate the problem:

- when a status response schema refers to a `#/definitions/X`
- and neither X nor the schema contain an `example`
- then no 'model schema' is shown

I'm not sure what other conditions are required, if any.

**Comment by ponelat** [comment]
hmm. that is definitely not the desired behaviour. Model Schema should always show something useful, with or without the examples set.

Although, I can't replicate the issue. I'm using the spec file above (latest from gh-pages), with the examples taken out (every single one) - but I get the expected values under Model Schema.
One note: I think you should put _all_ definitions under `definitions` and not use arbitrary names like `parameters`. This could mess with things, even though JSON Schema does support arbitrary names, we have references to `definitions` throughout the code.

So, if you could move the definitions under `parameters` to be under `definitions` and update to the latest (the optimal being develop_2.0 branch) and see if you attain the desired results.

I've tried the above commit hash, and using the latest.
Here is what I get (at develop_2.0) **I've taken out every instance of _example_**
![screenshot from 2015-06-02 13 22 54](https://cloud.githubusercontent.com/assets/8438485/7934729/7e1386a0-092a-11e5-97c5-7902596a1afb.png)

I've also noted that there may be cases where an enum would suit, you can take a look at the petstore example if it'll help.... https://github.com/swagger-api/swagger-spec/blob/master/examples/v2.0/yaml/petstore-expanded.yaml

**Comment by ponelat** [comment]
hey @stevage I'm must **humbly** apologize. I've led you astray with my own erroneous notions of the `parameters`. You are using just as the spec/swagger-js/swagger-ui designed it to be used. Here is the proof to back you up... https://github.com/swagger-api/swagger-spec/blob/master/guidelines/REUSE.md#parameters.

Regarding the actual issue, I haven't been able to replicate yet.
If you have this...
```yaml
 schema:
        title: Departures
        type: array
        items:
          $ref: "#/definitions/timetable_value"
```
You should see some dummy (although pretty plain) stuff under 'Model Schema'/

**Comment by webron** [comment]
@stevage - it looks like the issues raised by the OP are resolved and @ponelat cannot reproduce your problems. We'll gladly continue working on your issue if it exists, but please open a separate issue for it.

**Comment by stevage** [comment]
Thanks guys, I really appreciate your efforts on this. I had a really busy week and wasn't able to have another look at this. In the end, the workaround worked for my immediate purpose and I'm not sure when I'll be diving into it next.

My best guess at why you couldn't repro is I wasn't quite on master or something - sorry if I wasted your time.

Here's how my site ended up: http://www.opentransportdata.org/
